packages/react-noop-renderer/src/createReactNoop.js

```diff
@@ -832,38 +830,52 @@ function createReactNoop(reconciler: Function, useMutation: boolean) {
 
     interactiveUpdates: NoopRenderer.interactiveUpdates,
 
-    // maybe this should exist only in the test file
     act(callback: () => void): Thenable {
-      // note: keep these warning messages in sync with
-      // ReactTestRenderer.js and ReactTestUtils.js
-      let result = NoopRenderer.batchedUpdates(callback);
+      actDepth++;
+
+      const result = NoopRenderer.batchedUpdates(callback);
+
+      let didAwait = false;
+
       if (__DEV__) {
-        if (result !== undefined) {
-          let addendum;
-          if (result !== null && typeof result.then === 'function') {
-            addendum =
-              "\n\nIt looks like you wrote ReactNoop.act(async () => ...) or returned a Promise from it's callback. " +
-              'Putting asynchronous logic inside ReactNoop.act(...) is not supported.\n';
-          } else {
-            addendum = ' You returned: ' + result;
-          }
-          warningWithoutStack(
-            false,
-            'The callback passed to ReactNoop.act(...) function must not return anything.%s',
-            addendum,
-          );
-        }
-      }
-      ReactNoop.flushPassiveEffects();
-      // we want the user to not expect a return,
-      // but we want to warn if they use it like they can await on it.
-      return {
-        then() {
-          if (__DEV__) {
+        Promise.resolve()
+          .then(() => {})
+          .then(() => {
             warningWithoutStack(
-              false,
-              'Do not await the result of calling ReactNoop.act(...), it is not a Promise.',
+              didAwait,
+              'You called act() without awaiting its result.',
             );
+          });
+      }
+
+      return {
+        then(resolve, reject) {
+          if (didAwait) {
+            return;
+          }
+          didAwait = true;
+          if (
+            result !== undefined &&
+            result !== null &&
+            typeof result.then === 'function'
+          ) {
+            result.then(() => {
+              actDepth--;
+              if (actDepth === 0) {
+                // TODO: Should keep flushing until queue is empty
+                ReactNoop.unstable_flushWithoutYielding();
```

Comment:
this would simplify tests too (which I assume was the reason you did this). til, thanks!